services/billing.py

```diff
@@ -155,18 +157,64 @@ def delete_subscription(self, owner: Owner):
             f"Downgrade to basic plan from user plan for owner {owner.ownerid} by user #{self.requesting_user.ownerid}",
             extra=dict(ownerid=owner.ownerid),
         )
+
         if subscription_schedule_id:
             log.info(
                 f"Releasing subscription from schedule for owner {owner.ownerid} by user #{self.requesting_user.ownerid}",
                 extra=dict(ownerid=owner.ownerid),
             )
             stripe.SubscriptionSchedule.release(subscription_schedule_id)
 
-        stripe.Subscription.modify(
-            owner.stripe_subscription_id,
-            cancel_at_period_end=True,
-            proration_behavior="none",
+        # we give an auto-refund grace period of 24 hours for a monthly subscription or 72 hours for a yearly subscription
+        # current_subscription_timestamp = subscription["current_period_start"]
+        current_subscription_datetime = datetime.fromtimestamp(
+            subscription["current_period_start"]
         )
+        differenceFromNow = datetime.now() - current_subscription_datetime
+
+        subscription_plan_interval = (
```

Comment:
This is equivalent to subscription_plan_interval = getattr(subscription.plan, "interval", None) I believe